Geox 1

nicholascar opened the pull request:
The first series of updates to 1.1 from the [GeoSPARQL Extensions Ontology](http://linked.data.gov.au/def/geox).

I have also fixed a few errors here and there and updated all ont refs to the defining artefact as the Profile 1.1, not 1.0.

See the detailed Issue #33 's comment for reasoning: https://github.com/opengeospatial/ogc-geosparql/issues/33#issuecomment-753750117

dr-shorthair commented on 1.1/spec/09-Part-06.adoc:
```diff
@@ -11,7 +11,7 @@ This clause establishes the *core* requirements class, with URI `/req/core`, whi
 ==== Classes
 
-Two main classes are defined: `geo:SpatialObject` and `geo:Feature`. The class `geo:Feature` is equivalent to the UML class `GFI_Feature` defined in [8].
+Three main classes are defined: `geo:SpatialObject`, `geo:SpatialMeasure` and `geo:Feature`. The class `geo:Feature` is equivalent to the UML class `GFI_Feature` defined in [8].
```
"The class `geo:Feature` is equivalent to the UML class `Feature` defined in ISO 19109:2015."

nicholascar replied:
I've already addressed this one!

nicholascar replied:
Fixed with https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/17073e57566de6e6773919f1af6fed9cadde0cde

dr-shorthair commented on 1.1/spec/09-Part-06.adoc:
```diff
@@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
+                    owl:Class;
+    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>;
+    rdfs:label "Spatial Measure"@en;
+    rdfs:comment "The class represents a measurement of a feature bot only 
```
"but", maybe 'though"

nicholascar replied:
Fixed with https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/708975dba30cd71c1ec7165eb4e4d2d4f4c17e6c

dr-shorthair commented on 1.1/spec/09-Part-06.adoc:
```diff
@@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
+    rdfs:label "Spatial Measure"@en;
+    rdfs:comment "The class represents a measurement of a feature bot only 
+                spatial measurements. It is therefore naturally a subclass 
+                of Spatial Object which is any spatial thing."@en .
```
Not sure I agree with that. 'Measure' is from ISO 19103, and I think is a datatype, not a class. It has to be implemented in RDFS/OWL as a class because it has more than one component, but functionally it is a datatype for a scaled number.

nicholascar replied:
So what do you suggest? It is not then to be a part of the `Geometry` / `SpatialObject` hierarchy?

dr-shorthair replied:
Correct. It does not belong in that subsumption chain.

nicholascar replied:
OK, then how do we align the `hasArea` etc. with `hasGeometry`? Clearly `Feature hasGeometry Geometry` is similar to `Feature hasArea/hasLength/hasVolume SpatialMeasure` so what is the superclass of `Geometry` & `SpatialMeasure`? Is it to be left unstated in GeoSPARQL or is `SpatialObject` the superclass?

I'm worried about not tying all the property things together for surely soon we will have `Feature hasTime TemorpalObject` and we will have this all over again.

Should we bite the bullet and have, within GeoSPARQL, 

```
                     Measure
              ^                      ^
SpatialMeasure TemporalMeasure
     ^
Geometry
```

or similar? Or do we cut off GeoSPARQL at the Spatial? I'd prefer to include Temporal as I kinda need spatio-temporal objects.

dr-shorthair replied:
hasArea / hasLength etc is kinda like pointing to one component of a coordinate. It is an incomplete description. Useful, but incomplete. You might make both `hasGeometry` and `hasLength` sub-properties of `hasSpatialProperty` but IMO it isn't worth the effort. 

As I mentioned, ISO 19105 does not think Measure has a super-class

2005 version: 
![image](https://user-images.githubusercontent.com/608303/103599144-87dc2500-4f58-11eb-9a83-f859dcdb110d.png)

2013 revision: 
![image](https://user-images.githubusercontent.com/608303/103599201-b3f7a600-4f58-11eb-91a4-1f7d59bc99cd.png)

lvdbrink replied:
From issue #33: 
> it is required to keep the Geometry / SpatialObject relations unchanged. 

Do I understand correctly that this is the reason for introducing the `SpatialMeasure` class? Understandable, but this introduces a deviation from ISO 19107. That may not be a good idea. 

That said, from my limited understanding of ISO 19107 (not having a background in geodesy), a geometry is a measure, although the term Measure seems to be understood (in ISO 19105) to mean other (simpler) types of quantities.

mathib replied:
small note: during a design process (e.g. road design, building design, etc.) geometry can be designed instead of measured.

nicholascar replied:
> You might make both hasGeometry and hasLength sub-properties of hasSpatialProperty but IMO it isn't worth the effort.

I think it might be. I do think that a geometry is just a particular kind of measurement. It may be a compound one (lots of individual lengths/points etc.) but it's still just a measure (of spatiality) just as hasTime would be a measure of temporality.

I'm keen to see geometry brought into the measurement fold.

> geometry can be designed instead of measured

I don't think this is really a real difference! It's still a property of something (a Feature) as seen from GeoSPARQL's point of view.

mathib replied:
I think it's just a matter of wording: I agree that geometry is a special type of property of a FoI, but I would not call it a measurement. Are the `Measure` and `TemporalMeasure` classes already defined in some ontology? BTW: instead of directly discussing TBox, we can maybe start the discussion from data level (ABox)? In the OMG/FOG/GOM approach we connect geometry and related properties (incl. spatial ones such as length and area) as follows:
```
inst:road1 omg:hasGeometry inst:road1-geom1 ;
   ex:length inst:road1-prop1 .
inst:road1-geom1 a omg:Geometry ;
   fog:asObj_v3.0-obj "an OBJ string"^^xsd:string .
inst:road1-prop1 a opm:Property , seas:Property ;
   schema:value "100.4"^^xsd:decimal ;
   qudt:unit unit:M ;
   omg:isImplicitlyDerivedFrom inst:road1-geom1 .
```
The (currently fictive) spatial property `ex:length` is not a sibling of `omg:hasGeometry` as I don't see much advantages except for more clarity on an abstract level (the same goes for the `omg:Geometry` and `opm:Property` classes: the first is no subclass of the latter). In that sense, I agree with @dr-shorthair. Personally, I think that ABox links between the property and the geometry are more interesting, e.g. to indicate that the length property is measured on (derived from) the geometry object.

nicholascar replied:
> Are the Measure and TemporalMeasure classes already defined in some ontology?

We are exploring that here. `SpatialMeasure` in our extension ontology for GeosPARQL (https://linked.data.gov.au/def/geox) is a sub class of a general `Measure` class which is defined in a separate upper ontology. GeoSPARQL's upper classes are indeed the issue here: how far should GeoSPARQL go "up" in generalities?

> I agree that geometry is a special type of property of a FoI, but I would not call it a measurement

Why not? Isn't geometry a representation of a Feature's spatiality - it's particular value for a property from a set of all possible values, just as a Feature's  length is from all possible lengths or its temperature from all temperatures? What's so special about a Feature's spatial presence that it's not just another abstraction of a property of a thing?

> instead of directly discussing TBox, we can maybe start the discussion from data level (ABox)

ABox to illustrate potential instances, sure. But the ABox you are showing is from elsewhere and only represent that scenario's TBox in use, not what could be in use here.

> The (currently fictive) spatial property ex:length is not a sibling of omg:hasGeometry

Well perhaps you don't have it that way yet but the the position here is that it could be. Saying it currently isn't now is no argument for or against what it could be.

> I don't see much advantages except for more clarity on an abstract level

I am concerned about the orderliness of the representations of dimensions of things so I care about this issue of whether properties - spatial and others - can be aligned in their representation structure vis a vis a thing. I think that this is important for the complexity scaling of models (when things like temporality are added). If spatiality and other properties can be aligned they should be. If alignment causes problems, they shouldn't be. 

We can't know all possible future uses of these dimensions of things but we should try and aim for cohesion of representation, unless it proves problematic.

> Personally, I think that ABox links between the property and the geometry are more interesting, e.g. to indicate that the length property is measured on (derived from) the geometry object.

Sure, that _may_ be interesting but it doesn't aid or prevent the TBox modelling one way or the other. Measures could be derived from other measures, for example an average temperature of a Feature taken from a range of temperatures or, back to spatial land, a centroid from a polygon.

mathib replied:
Hi @nicholascar , thanks for all the pointers and elaborate explanations :). Noted that Geox is documented nicely with ABox examples.

I think I'm just a little distracted by the word "measurement", which I connect mentally to something that is measured physically, e.g. as used in the sentence "_I measure the width of the road using a total station_". In the construction domain a division is often made between such measured properties (as-is or as-built) vs designed properties (as-designed, meaning the road might not exist yet so there's nothing you can physically measure. As a designer you can declare that the width is equal to a certain value.). 

If the group decides to include `SpatialMeasurement` (I would go for `SpatialProperty` but maybe I'm the only one distracted by measurement) as a superclass for `Geometry`, is it then possible to add the above as a note/description to this superclass?

dr-shorthair replied:
Measure/measurement are rather fraught, can be both verbs and nouns :-( 
I snipped those diagrams from ISO 19103 primarily to clarify to the team that there is a class (datatype) `Measure` in our ecosystem, which is simply for scaled numbers, and the class `SpatialMeasure` in GEOX was modelled on that.

dr-shorthair replied:
(I know that was the intention because I created it in GEOX.)

nicholascar replied:
I'll resolve this comment to hide it in this Conversation but noted that there's an "If the group decides..." above so we might have to follow-up the output of this PR with another.

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
```diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
-====  Standard Properties for geo:Geometry
+===== Property: geo:hasLength
+
+The property `geo:hasLength` is used to indicate the length of a `geo:Feature`. The property indicates `geo:SpatialMeasure` object which then contains the length value expressed as a scalar number and which potentially also includes units of measure, uncertainty and other properties.
```
2nd sentence: 

The range of the property is `geo:SpatialMeasure`, which encodes the length value expressed as a scalar quantity and which also includes the units of measure, and potentially uncertainty and other properties.

nicholascar replied:
Fixed with https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/b79b1d5da4562d0c9dc13aa914d440eb60618534

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
````diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
+    rdfs:range geo:SpatialMeasure .
+```
+
+An ontological consistency check can be applied to geometries indicating this property and the `geo:dimension` property: the integer range value of a `geo:dimension` property for a geometry with a `geo:hasLength` property must be 1.
````
"An ontological consistency check" --> "A consistency check"

nicholascar replied:
Fixed by https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/1e37e0c1ba61f045a58a3b2f7925b1c7cdb7c661

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
```diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
+
+===== Property: geo:hasArea
+
+The property `geo:hasArea` is used to indicate the area of a `geo:Feature`. The property indicates `geo:SpatialMeasure` object which then contains the area value expressed as a scalar number and which potentially also includes units of measure, uncertainty and other properties.
```
See comment on l108

nicholascar replied:
Fixed with https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/b79b1d5da4562d0c9dc13aa914d440eb60618534

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
```diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
+
+===== Property: geo:hasVolume
+
+The property `geo:hasVolume` is used to indicate the area of a `geo:Feature`. The property indicates `geo:SpatialMeasure` object which then contains the volume value expressed as a scalar number and which potentially also includes units of measure, uncertainty and other properties.
```
See comment on l108

nicholascar replied:
Fixed with https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/b79b1d5da4562d0c9dc13aa914d440eb60618534

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
````diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
+    rdfs:range geo:SpatialMeasure .
+```
+
+An ontological consistency check can be applied to geometries indicating this property and the `geo:dimension` property: the integer range value of a `geo:dimension` property for a geometry with a `geo:hasLength` property must be 2.
````
See comment on l120

nicholascar replied:
Fixed by https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/1e37e0c1ba61f045a58a3b2f7925b1c7cdb7c661

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
````diff
@@ -103,12 +103,60 @@ WHERE { :f1 geo:sfDisjoint :f1 }
+    rdfs:range geo:SpatialMeasure .
+```
+
+An ontological consistency check can be applied to geometries indicating this property and the `geo:dimension` property: the integer range value of a `geo:dimension` property for a geometry with a `geo:hasLength` property must be 3.
````
See comment on l120

nicholascar replied:
Fixed by https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/1e37e0c1ba61f045a58a3b2f7925b1c7cdb7c661

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
```diff
@@ -197,15 +258,30 @@ The `geo:hasSerialization` property is used to connect a geometry with its text-
 
+===== Property: geo:inCRS
+
+The `geo:inCRS` property is used to connect a geometry with its text- based serialization (e.g., its WKT serialization).
```
"The `geo:inCRS` property is used to connect a geometry with a text- based serialization of the CRS used for its representation (e.g. a WKT serialization)."

situx replied:
Would this definition also include EPSG codes/EPSG URIs/SRID numbers? Or is there another property for this?

nicholascar replied:
Fixed by https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/d0d40f5fcce89a516a7314b3048879f7ddf5814c

> Would this definition also include EPSG codes/EPSG URIs/SRID numbers? Or is there another property for this?

I've set the range value for the property to `skos:Concept`, so the object has to be a URI then, rather than a literal or even a microformat like a `dcterms:identifier` / `skos:notation`. I've asked the OGC already to update all their EPSG URIs like <http://www.opengis.net/def/crs/EPSG/0/4326> which currently only return XML, to returning Linked Data, so HTML & RDF as well as the current XML. When they do, those URIs will fit perfectly for use here. 

The OGC's data in that `/def/crs` resource is fine, it's just the presentation that's lacking...

We will surely too define some SHACL shapes for this 1.1 profile to apply further checks for appropriate values for things like this `inCRS` property.

dr-shorthair commented on 1.1/spec/11-Part-08.adoc:
```diff
@@ -197,15 +258,30 @@ The `geo:hasSerialization` property is used to connect a geometry with its text-
+            owl:ObjectProperty;
+    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>; 
+    rdfs:label "in CRS"@en;
+    rdfs:comment "A Coordinate Reference System, as recorded in a 
```
"The coordinate reference system of the coordinates in the representation of the geometry."

nicholascar replied:
Fixed by https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/4015f72abbff6c2e4dfe2ef16ca3f97924f13732

dr-shorthair commented on 1.1/geo.ttl:
```diff
@@ -231,26 +232,101 @@ xsd:date a rdfs:Datatype .
+	dc:contributor "Simon J.D. Cox / Nicholas J. Car" ;
+	dc:creator "OGC GeoSPARQL Standard Working Group" ;
+	dc:date "2020-01-04"^^xsd:date ;
+	dc:description """The length of a one-dimensional Feature, expressed as a Spatial Measure."""@en ;
```
Does length only apply to 1-D features?

nicholascar replied:
I did think so but perhaps you are thinking of perimeter of a 2D think length?

Applied commit https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/13ce7a17081b325872b4b598c69e5f44f88cd7b0, does that cover it?

dr-shorthair replied:
Length is commonly used to denote the extent of the longest side or dimension. Length of a swimming pool, or of a football field, lake, peninsula, etc. Perhaps such common semantics is unhelpful here, but I wanted to ask the question at least.

nicholascar replied:
I'm ok with a multi-dimentional possible interpretation of length (or any other property) but I think that "Length ... used to denote the extent of the longest side or dimension" is a special (Quantity Kind?) length and is really just length but pertaining to something in particular - still length. Sounds like a property with qualifiers, roles, etc., or even a different target Feature.

kjano commented:
> OK, then how do we align the |hasArea| etc. with |hasGeometry|? 
> Clearly |Feature hasGeometry Geometry| is similar to |Feature 
> hasArea/hasLength/hasVolume SpatialMeasure| so what is the superclass 
> of |Geometry| & |SpatialMeasure|? Is it to be left unstated in 
> GeoSPARQL or is |SpatialObject| the superclass?

This raises many other interesting issues such as the relation between a 
SpatialMeasure an a SSN Observation. Also hasArea, hasLenght, has Volume 
and so on should probably point to *results* of 
measurements/observations, correct, i.e., not events?

Jano

On 1/4/21 6:08 PM, Nicholas Car wrote:
>
> *@nicholascar* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In 1.1/spec/09-Part-06.adoc 
> <https://github.com/opengeospatial/ogc-geosparql/pull/72#discussion_r551676012>:
>
> > @@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
>   |`/req/core/spatial-object-class`
>   |===
>   
> +===== Class: geo:SpatialMeasure
> +
> +The class `geo:SpatialMeasure` is defined by the following:
> +
> +```
> +geo:SpatialMeasure a rdfs:Class,
> +                    owl:Class;
> +    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>;
> +    rdfs:label "Spatial Measure"@en;
> +    rdfs:comment "The class represents a measurement of a feature bot only
> +                spatial measurements. It is therefore naturally a subclass
> +                of Spatial Object which is any spatial thing."@en .
>
> OK, then how do we align the |hasArea| etc. with |hasGeometry|? 
> Clearly |Feature hasGeometry Geometry| is similar to |Feature 
> hasArea/hasLength/hasVolume SpatialMeasure| so what is the superclass 
> of |Geometry| & |SpatialMeasure|? Is it to be left unstated in 
> GeoSPARQL or is |SpatialObject| the superclass?
>
> I'm worried about not tying all the property things together for 
> surely soon we will have |Feature hasTime TemorpalObject| and we will 
> have this all over again.
>
> Should we bite the bullet and have, within GeoSPARQL,
>
> |Measure ^ ^ SpatialMeasure TemporalMeasure ^ Geometry |
>
> or similar? Or do we cut off GeoSPARQL at the Spatial? I'd prefer to 
> include Temporal as I kinda need spatio-temporal objects.
>
> —
> You are receiving this because you are subscribed to this thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/opengeospatial/ogc-geosparql/pull/72#discussion_r551676012>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AANMP5W33F2SFHASCYH4O6DSYJYCFANCNFSM4VS7YYAA>.
>

-- 
Krzysztof Janowicz

Geography Department, University of California, Santa Barbara
4830 Ellison Hall, Santa Barbara, CA 93106-4060

Email: jano@geog.ucsb.edu
Webpage: http://geog.ucsb.edu/~jano/
Semantic Web Journal: http://www.semantic-web-journal.net

kjano commented:
On 1/4/21 6:20 PM, Simon Cox wrote:
>
> *@dr-shorthair* commented on this pull request.
>
> ------------------------------------------------------------------------
>
> In 1.1/spec/09-Part-06.adoc 
> <https://github.com/opengeospatial/ogc-geosparql/pull/72#discussion_r551679329>:
>
> > @@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
>   |`/req/core/spatial-object-class`
>   |===
>   
> +===== Class: geo:SpatialMeasure
> +
> +The class `geo:SpatialMeasure` is defined by the following:
> +
> +```
> +geo:SpatialMeasure a rdfs:Class,
> +                    owl:Class;
> +    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>;
> +    rdfs:label "Spatial Measure"@en;
> +    rdfs:comment "The class represents a measurement of a feature bot only
> +                spatial measurements. It is therefore naturally a subclass
> +                of Spatial Object which is any spatial thing."@en .
>
> hasArea / hasLength etc is kinda like pointing to one component of a 
> coordinate. It is an incomplete description. Useful, but incomplete.
>

Can you explain this in a bit more details? I certainly agree on the 
case of coordinate components but, hasArea and hasLength seem fine on 
their own.

Thanks.

Jano

> You might make both |hasGeometry| and |hasLength| sub-properties of 
> |hasSpatialProperty| but IMO it isn't worth the effort.
>
> As I mentioned, ISO 19105 does not think Measure has a super-class
>
> 2005 version:
> image 
> <https://user-images.githubusercontent.com/608303/103599144-87dc2500-4f58-11eb-9a83-f859dcdb110d.png>
>
> 2013 revision:
> image 
> <https://user-images.githubusercontent.com/608303/103599201-b3f7a600-4f58-11eb-91a4-1f7d59bc99cd.png>
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub 
> <https://github.com/opengeospatial/ogc-geosparql/pull/72#discussion_r551679329>, 
> or unsubscribe 
> <https://github.com/notifications/unsubscribe-auth/AANMP5QWLUFLKIGYK4EHO2LSYJZORANCNFSM4VS7YYAA>.
>

-- 
Krzysztof Janowicz

Geography Department, University of California, Santa Barbara
4830 Ellison Hall, Santa Barbara, CA 93106-4060

Email: jano@geog.ucsb.edu
Webpage: http://geog.ucsb.edu/~jano/
Semantic Web Journal: http://www.semantic-web-journal.net

dr-shorthair commented:
@kjano I mean length/area/volume provides a single (scaled) number to describe an aspect of a spatial thing. Components of a coordinate are single (scaled/offset) numbers that describe other aspects. Sometimes a single measure is useful for some application, but none of these is 'complete' without a whole lot of other numbers. Maybe the analogy is a bit tortured. But my point is that a single coordinate-component is not a 'spatial thing' and a measure of its 1D/2D/3D extent is not usefully 'spatial' in my opinion. It is just a measure. Measures only become 'spatial' when combined with other measures, and embedded in some N-dimensional space.

lvdbrink reviewed (changes requested):
Some minor comments:
- Labels of the properties are not consistent, e.g. “has default geometry” vs “coveredBy”. 
- copy errors in labels, e.g. hasArea rdfs:label “hasGeometry”, same for inCRS
- the class def of :SpatialObject is preceded by “#http://www.opengis.net/ont/geosparql#SpatialMeasure” 
- The dc:description of :SpatialObject says that it is superclass of feature and geometry, but as of 1.1 it is also superclass of spatial measure.
- Topology Vocabulary Extension Clause: The abbreviations of geometry types in the tables should be explained.
- in part 08 property domain doesn’t seem right for :inCRS rdfs:domain geo:SpatialMeasure;
- in part 08 some requirements are not numbered yet
- in part 08 some inline code styles do not work properly (when they are followed by ‘s’ without space)
- in part 10 requirements for GeoJSON and KML serialization seem to be missing. See also https://github.com/opengeospatial/ogc-geosparql/issues/62.
- part 11, RIF rules. I haven’t seen RIF in common use. Not sure how useful this clause is.

nicholascar commented:
@lvdbrink, I have addressed all your points here with commit https://github.com/opengeospatial/ogc-geosparql/pull/72/commits/d88570eaed0a2f40153953c281749d4ccf877b0c with most being resolved, one or two deferred and a couple out-of-scope

* Labels of the properties are not consistent, e.g. “has default geometry” vs “coveredBy”.
    * in this PR I've only tackled the specifics of GeoX, not patched other things like labels with bad styling to keep the diff down. I have supplied a separate PR for that, see PR #79 
* copy errors in labels, e.g. hasArea rdfs:label “hasGeometry”, same for inCRS
    * fixed for hasArea & inCRS. But also the duplicative annotations for `rdfs:label` & `skos:prefLabel` are tided up n PR #79
* the class def of :SpatialObject is preceded by #SpatialMeasure
    * fixed 
* The dc:description of :SpatialObject says that it is superclass of feature and geometry, but as of 1.1 it is also superclass of spatial measure.
    * I've un-entangled this so SpatialObject is, as per GeoSPARQL 1.0, the superclass of just :Feature & :Geometry. :SpatialMeasure has no superclasses defined
* Topology Vocabulary Extension Clause: The abbreviations of geometry types in the tables should be explained.
    * I've started a new Issue to track things that still need definitions: https://github.com/opengeospatial/ogc-geosparql/issues/81
    * I've also added full names for _Egenhofer_, _DE-9IM_ & _RCC8_
* in part 08 property domain doesn’t seem right for :inCRS rdfs:domain geo:SpatialMeasure;
    * I've converted the domain to `geo:Geometry`, given the changes to un-entangle SpatialMeasure noted above
* in part 08 some requirements are not numbered yet
    * this will have to be fixed post this PR since there are other requirements coming in in other PRs
* in part 08 some inline code styles do not work properly (when they are followed by ‘s’ without space)
    * fixed with use of `xxx` instances
* in part 10 requirements for GeoJSON and KML serialization seem to be missing. See also #62.
    * out of scope for this PR so leaft to handing of #62 and other PRs
* part 11, RIF rules. I haven’t seen RIF in common use. Not sure how useful this clause is.
    * out of scope for this PR so leaft to handing of #62 and other PRs

lvdbrink commented:
Looks good!

situx commented:
Thanks for the good work. Just one question: If the inCRS property is defined and a CRS URI is defined in a WKTLiteral. Which CRS definition will take precedence? The literal or the CRS defined in inCRS?

mathib commented:
+ GeoJSON with its fixed CRS (WGS84): in that case, the CRS from `inCRS` should probably be ignored? Don't know about KML?

situx commented:
Yes, it would probably be this way, but I recommend that we make this explicit in the standard.

nicholascar commented:
> If the inCRS property is defined and a CRS URI is defined in a WKTLiteral.

Actually, WKT _does not_ allow for CRS/SRID definitions, as far as I can tell. It's only Postgres' extension that does with `SRID=...;` at the start of the literal (see [Wikipedia](https://en.wikipedia.org/wiki/Well-known_text_representation_of_geometry#Format_variations).

But the general point about what to do with a inCRS / literal-defined CRS collision stands.

I think we should take the WG's opinion on this. I usually look for which property would require the most deliberate human action to set and, I think, that's going to be `inCRS` but I can see it makes little sense to ignore the CRS as set in a literal along with the coordinate data. So I suspect we will choose literal overrides `inCRS`, as you two have discussed above, and indicate in documentation that `inCRS` is mainly for use without a literal declaration and we will need to provide guidance in the ontology (and example) and the spec for this. I have created an Issue for it: Issue #82.

> GeoJSON with its fixed CRS (WGS84): in that case, the CRS from inCRS should probably be ignored? Don't know about KML?

We might soon have to make a new literal for GeoJSON v2 if it allows CRS but v1 (current) doesn't. Or we can assume that a GeoJSOn with a CRS indicated is GeoJSON v2.

mathib commented:
Great, we can move this part of the discussion to the new issue #82 !

For the CRS **URIs** in general, which ones should be used? Do users have the option to deviate from what will be defined in the GeoSPARQL spec? (e.g. are custom CRS allowed, even though they might not be recognized by a processing tool?). For ref, GeoSPARQL 1.0 only says the following for WKT:

> The OGC maintains a set of CRS URIs under the http://www.opengis.net/def/crs/ namespace.

I read this as a (strong) suggestion to use URIs under that namespace only, but not as an obligation... Do we need to make it more explicit in the new spec? In any case, the suggestion/obligation for CRS URIs should also be included in the ontology's term description.

In addition, will it be allowed to point `inCRS` to a 3D CRS (from which an application can recognize the underlying 2D CRS needed for 2D spatial querying as currently supported by GeoSPARQL 1.0 and the upcoming 1.1)? KML, WKT and GML allow 3D coordinates (GeoJSON currently only 2D), so we might need to explicitly state if 3D coordinates are allowed or not in GeoSPARQL 1.1? Actually, I believe KML only supports 3D coordinates?

situx commented:
I understand that the CRS URI is completely arbitrary because there might be different providers of CRS definitions, thus allowing different URI specifications to be applicable in CRS URIs. If we were to restrict the URI then we would also reject other providers of CRS definitions apart from OGC. I think that is too restrictive.

I always understood that if literal types are able to represent 3D coordinates or even 4D coordinates they may be represented in the GeoSPARQL standard. Many of the query function descriptions are ambiguous enough to also not reject 3D coordinate geometries, even though implementations may work on the 2D representations ignoring the Z coordinate.

But on the issue of including 3D support there is a ticket which is to be done for GeoSPARQL anyway, so maybe we should take a look at that in the next meeting?

kjano commented:
> @kjano I mean length/area/volume provides a single (scaled) number to describe an aspect of a spatial thing. Components of a coordinate are single (scaled/offset) numbers that describe other aspects. Sometimes a single measure is useful for some application, but none of these is 'complete' without a whole lot of other numbers. Maybe the analogy is a bit tortured. But my point is that a single coordinate-component is not a 'spatial thing' and a measure of its 1D/2D/3D extent is not usefully 'spatial' in my opinion. It is just a measure. Measures only become 'spatial' when combined with other measures, and embedded in some N-dimensional space.

@dr-shorthair, @nicholascar I am looking at this again and still agree (and like) the idea of expressing areas, length, and so on as results of observations. This is great because it gives us a nice framework to also model how we arrived at these values and so on. The same can be said about geometries. I also understand why we would want to name these spatial measures. What I hope could be discussed in more detail is how this relates to observations. If observations (in O&M and SSN/SOSA) are events, would spatial measures also be events or would they be the results of observations? In both cases, it would be unclear why SpatialMeasure would be a subclass of SpatialObject.

dr-shorthair commented:
Since this PR has already been merged, we should continue this discussion in a new issue - see #83

dr-shorthair commented:
Nevertheless, I disagree that a spatial-measure is a sub-class of spatial-object, which is stated in the documentation of spatial-measure (though it does not appeared to be axiomatized).

dr-shorthair commented on 1.1/spec/09-Part-06.adoc:
```diff
@@ -32,6 +32,25 @@ geo:SpatialObject a rdfs:Class,
+    rdfs:isDefinedBy <http://www.opengis.net/spec/geosparql/1.1>;
+    rdfs:label "Spatial Measure"@en;
+    rdfs:comment "The class represents a measurement of a feature though only 
+                spatial measurements. It is therefore naturally a subclass 
```
I disagree that SpatialMeasure is a _subclass_ of spatial object. It behaves like a data-type, not a class of objects in the world.

nicholascar replied:
This is an error: the subclassing is removed in the ont,but seemingly not in the Spec doc. Will PR to fix:

* [SpatialMeasure in the ont](https://github.com/opengeospatial/ogc-geosparql/blob/master/1.1/geo.ttl#L511)

nicholascar replied:
Actually, this is solved in the Spec master version, see https://github.com/opengeospatial/ogc-geosparql/blob/master/1.1/spec/09-Part-06.adoc so I think this can be closed.
